examples/upgrade/blue-green-route53/core-infra/main.tf

```
@@ -0,0 +1,115 @@
+provider "aws" {
+  region = local.region
+}
+
+data "aws_availability_zones" "available" {}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.0"
+
+  name = local.name
+  cidr = local.vpc_cidr
+
+  azs             = local.azs
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k)]
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 10)]
+
+  enable_nat_gateway   = true
+  create_igw           = true
+  enable_dns_hostnames = true
+  single_nat_gateway   = true
+
+  # Manage so we can name
+  manage_default_network_acl    = true
+  default_network_acl_tags      = { Name = "${local.name}-default" }
+  manage_default_route_table    = true
+  default_route_table_tags      = { Name = "${local.name}-default" }
+  manage_default_security_group = true
+  default_security_group_tags   = { Name = "${local.name}-default" }
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.name}-blue"     = "shared"
+    "kubernetes.io/cluster/${local.name}-green"    = "shared"
+    "kubernetes.io/cluster/${local.name}-nodomain" = "shared"
+    "kubernetes.io/role/elb"                       = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${local.name}-blue"     = "shared"
+    "kubernetes.io/cluster/${local.name}-green"    = "shared"
+    "kubernetes.io/cluster/${local.name}-nodomain" = "shared"
+    "kubernetes.io/role/internal-elb"              = "1"
+  }
```

Are these tags still required? Afaik only the `"kubernetes.io/role/elb" = "1"` is required.

```suggestion
  public_subnet_tags = {
    "kubernetes.io/role/elb"                       = "1"
  }

  private_subnet_tags = {
    "kubernetes.io/role/internal-elb"              = "1"
  }
```

---

i'll test this

---

So yes this is needed for Karpenter, we uses it in the subnet selector
```
  subnetSelector:
    kubernetes.io/cluster/{{ .Values.spec.clusterName }}: '*'
    kubernetes.io/role/internal-elb: '1' # to select only private subnets
```

Without this, your Karpenter nodes could be created in another VPC used maybe by another cluster